OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -43,17 +43,16 @@ static class DMOpcodeHandlers {
                 }
             }
 
-            //Enumerate a copy of the list
-            List<DreamValue> values = new();
-            if (list != null) {
-                values.Capacity = list.GetLength();
-
-                foreach (DreamValue value in list.GetValues()) {
-                    values.Add(value);
-                }
+            if (list == null)
+            {
+                List<DreamValue> values = new List<DreamValue>();
+                state.EnumeratorStack.Push(values.GetEnumerator());
```

Comment:
Is this part necessary or can I just not `Push()` if it's null?

Reply:
An enumerator has to be pushed. You could do `Enumerable.Empty<DreamValue>().GetEnumerator()` instead.